test/shopify-cli/project_test.rb

```diff
@@ -40,5 +40,11 @@ def test_write_includes_identifiers
       ShopifyCli::Project.write(@context, :node, { 'other_option' => true })
       assert Project.current.config['other_option']
     end
+
+    def test_project_name_returns_last_entry_in_working_directory
+      Dir.stubs(:pwd).returns("/Users/john/my_app")
```

Comment:
I've had experiences recently where stubs like this on the file system caused problems in other tests, but is it safe to assume that wouldn't be a concern here?

Reply:
This also assumes that the Project method will use Dir.pwd, which can cause some surprises if the method is ever refactored.

I believe there is a plan (or an issue) for adding better protection against system calls in the tests, so this may not be a problem for long.

Reply:
Yeah, it's not ideal, but we're already doing it in several other places in this class.

A more stable approach could be wrap it in an adapter and stub that.